Modules/_testcapimodule.c

```diff
@@ -1921,6 +1921,28 @@ unicode_asucs4(PyObject *self, PyObject *args)
     return result;
 }
 
+static PyObject *
+unicode_asutf8(PyObject *self, PyObject *args)
+{
+    PyObject *unicode, *result;
+    const char *buffer;
+
+    if (!PyArg_ParseTuple(args, "U", &unicode)) {
+        return NULL;
+    }
+
+    buffer = PyUnicode_AsUTF8(unicode);
+    if (buffer == NULL) {
+        return NULL;
+    }
+
+    result = PyUnicode_FromString(buffer);
+    if (result == NULL) {
```

Comment:
This explicit check for `result == NULL` is not necessary.
Just do `return func_name(buffer);`.

Reply:
Zackery, good catch. As Victor said,  I would use `PyBytes_FromString` :)